var/spack/repos/builtin/packages/zlib/package.py

```diff
@@ -40,7 +40,15 @@ class Zlib(AutotoolsPackage):
 
     variant('pic', default=True,
             description='Produce position-independent code (for shared libs)')
+    variant('shared', default=True,
+            description='Enables the build of shared libraries.')
 
     def setup_environment(self, spack_env, run_env):
         if '+pic' in self.spec:
             spack_env.set('CFLAGS', self.compiler.pic_flag)
+
+    def configure(self, spec, prefix):
```

Comment:
that is not the best way to do it, it's an `AutotoolsPackage`, you can add extra (non-prefix) arguments in 
```
def configure_args(self):
  config_args =[]
   if '+shared' not in self.spec:
       config_args.append('--static')
   return config_args
```